src/OrchardCore.Modules/OrchardCore.Workflows/Controllers/WorkflowTypeController.cs

```diff
@@ -121,7 +121,7 @@ public async Task<IActionResult> Index(WorkflowTypeIndexOptions options, PagerPa
                 .Take(pager.PageSize)
                 .ListAsync();
 
-            using var connection = await _session.CreateConnectionAsync();
+            var connection = await _session.CreateConnectionAsync();
```

Comment:
The name `CreateConnectionAsync` is really confusing because it will return the existing one (associated with the session) if it was used before (there was a query).

So by Disposing it we are creating the bug. Good catch. Would you mind checking the rest of the codebase for such usage patterns? And maybe add a comment here with your change that explains the connection already exists in the session and should not be disposed?

Reply:
It seems that there is no any other place with such a call of `session.CreateConnection` along with `using` keyword.